### ISSUE TITLE
There is a bug in the rendering order of Astro components

### ISSUE THREAD

**Issue by J-env** [withastro/astro#11475]
### Astro Info

```block
Astro                    v4.11.5
Node                     v18.18.0
System                   macOS (x64)
Package Manager          npm
Output                   static
Adapter                  none
Integrations             astro-icon
```

### If this issue only occurs in one browser, which browser is a problem?

Chrome

### Describe the Bug

When the same component uses `Astro.request` to record the first rendering state, but the first recorded state is not the frontmost component

```astro
---
import { Icon } from "astro-icon/components";
---

// Nav.astro
<nav>
 <Icon name="mdi:account" />
</nav>

// Footer.astro
<footer>
 <Icon name="mdi:account" />
</footer>
```

Rendering results:
<img width="760" alt="image" src="https://github.com/user-attachments/assets/f46d2063-5a5b-42a8-94f8-c8dce79ac73b">

### What's the expected result?

The example in the link is just a reproducible situation, it seems like there are other situations where this problem may also occur, perhaps related to slot rendering?

 `<symbol />` should appear as the first icon on the page, otherwise the first icon will be blank when the page is loaded.

Astro Icon: [Automatically optimized sprites](https://www.astroicon.dev/guides/components/#automatically-optimized-sprites)

<img width="646" alt="image" src="https://github.com/user-attachments/assets/da3909fc-f081-4464-98a6-52a7773a0efa">

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/withastro-astro-8pstxv?file=src%2Fpages%2Findex.astro

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by TheElegantCoding** [comment]
maybe this is related i hace a component that simulates a for each and render components it works like this.

```tsx
---
type Props<T> =
{
  each: T | null | false | undefined;
};

const { each } = Astro.props;

const slots: string[] = [];

if(each)
{
  await Promise.all(each.map(async (element: string, index: number) =>
  {
    slots.push(await Astro.slots.render('default', [ element, index ]));
  }));
}
---
<Fragment set:html={slots} />
```

this just put all slots or childrens and render the components, but after the new updates, the maps is not in order. this is how you use the component

```tsx
---
const data = [ '1', '2', '3', '4', '5', '6' ];
---

<For each={data}>
  {(data: string) => <p>{data}</p>}
</For>
```

the order of how this render change in dev sometimes and when i do the build also change the order one more time

**Comment by matthewp** [comment]
I don't think this is a bug. I'm not sure of anywhere that we say that components render in a top-down order.

**Comment by J-env** [comment]
> I don't think this is a bug. I'm not sure of anywhere that we say that components render in a top-down order.

But it affects some issues that are difficult to understand.
For example: Astro Icon records the first rendering, `<symbol />` should be rendered first.

**Comment by TheElegantCoding** [comment]
@matthewp I think it is, components should render in order, otherwise, this can affect the UI and how it looks

**Comment by NicholasEwing** [comment]
@matthewp Any update on this?

**Comment by matthewp** [comment]
This is intentional, we render in parallel so it can be done faster. You should not rely on global state.
